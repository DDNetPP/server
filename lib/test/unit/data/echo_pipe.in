started script at 2021-01-04_19-17-26
crash or shutdown2021-01-04_19-17-26
crash or shutdown2021-01-04_19-17-29
crash or shutdown2021-01-05_20-18-30
this line contains a " quote
