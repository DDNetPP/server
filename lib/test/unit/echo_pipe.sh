#!/bin/bash

if [ ! -f lib/lib.sh ]
then
    echo "Error: lib/lib.sh not found!"
    echo "make sure you are in the root of the server repo"
    exit 1
fi

source lib/lib.sh

log -n "echo_pipe.sh ... "

if ! diff <(./lib/echo_pipe.sh < ./lib/test/unit/data/crashes.txt) ./lib/test/unit/data/crashes_echo.txt
then
    echo "FAIL"
    err "echo pipe failed"
    exit 1
else
    echo "OK"
fi

