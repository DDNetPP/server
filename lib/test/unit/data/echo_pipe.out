echo "started script at 2021-01-04_19-17-26"
echo "crash or shutdown2021-01-04_19-17-26"
echo "crash or shutdown2021-01-04_19-17-29"
echo "crash or shutdown2021-01-05_20-18-30"
echo "this line contains a \" quote"
